include/openssl/comp.h

```diff
@@ -56,6 +56,16 @@ const BIO_METHOD *BIO_f_brotli(void);
 const BIO_METHOD *BIO_f_zstd(void);
 # endif
 
+typedef struct ssl_comp_st {
+    int id;
+    const char *name;
+    COMP_METHOD *method;
+} SSL_COMP;
+
+STACK_OF(SSL_COMP);
+
+DEFINE_STACK_OF(SSL_COMP)
```

Comment:
This must be generated through the `generate_stack_macros("SSL_COMP")` perl generator by moving that from ssl.h.in -> i.e., it requires generating comp.h from comp.h.in

Reply:
I'm wondering why we are bothering to move it? Why not just keep it in ssl.h.in?

Reply:
You need to be able to release the stack from libcrypto, so that means the definition needs to be in a header that code linked into libcrypto can include. IMO ssl.h should not be included.

Reply:
> IMO ssl.h should not be included.

Why not if we're only using it for some typedefs etc? Moving things to comp.h is problematic because it is excluded in the case of a no-comp build - whereas `SSL_COMP` should not be.

Reply:
the updated PR includes all suggestions from t8m. the code in `comp.h.in` got reshuffled so only relevant portion in that file is protected by OPENSSL_NO_COMP guard. I did test and build it:
    - perl ./Configure --strict-warnings enable-zlib enable-comp 
    - perl ./Configure --strict-warnings no-zlib no-brotli no-zstd no-comp 
 both versions did pass build and `make test`